docs/creating-a-static-check.md

```diff
@@ -0,0 +1,201 @@
+# Creating a Static Check
+
+Static checks are used to perform static analysis against a plugin's codebase.
+
+Static checks analyze the code, either using PHP CodeSniffer sniffs or custom logic e.g. using regular expressions.
```

Comment:
Would be good to mention the `Static_Check` interface here, as currently it's going right to the code example without that context. You can use the same explanation you have in the runtime check docs.
```suggestion
Static checks analyze the code, either using PHP CodeSniffer sniffs or custom logic e.g. using regular expressions.

The Plugin Checker provides the `Static_Check` interface, which is used to identify a static check. This interface does not contain any methods but serves as a marker for static checks.
```